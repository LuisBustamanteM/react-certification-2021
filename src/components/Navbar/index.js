export { default } from './Navbar.component.jsx';