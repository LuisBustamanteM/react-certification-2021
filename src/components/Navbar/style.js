import {colors} from '../../globalStyle';
import styled from 'styled-components';

export const NavBar = styled.div`
    margin: 0;
    text-align: left;
    color: ${colors.black};
    display:inline-flex;
    display: -webkit-flex;
    flex-wrap: wrap;
    justify-content: flex-start;
    position: relative;
    padding: 10px 24px 0 24px;
    background: ${colors.navBarBlue};
    min-height: 64px;
    box-shadow: 2px 2px 2px rgba(0,0,0,0.6);
`;