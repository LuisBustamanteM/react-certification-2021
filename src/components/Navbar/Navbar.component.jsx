import React, {useContext} from  'react';

import LeftNavComponent from '../LeftNav';
import RightNavComponent from '../RightNav';
import {StateContext} from "../App/App.component";

import {NavBar} from './style';

const NavbarComponent = () => {
    const {darkMode} = useContext(StateContext)

    return(
        <NavBar darkMode={darkMode}>
            <LeftNavComponent />
            <RightNavComponent/>
        </NavBar>
    )
}

export default NavbarComponent;