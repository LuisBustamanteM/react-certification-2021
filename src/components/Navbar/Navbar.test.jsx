import React from 'react';

import Navbar from './index';
import {render} from "@testing-library/react";

describe("Testing <Navbar/>", () => {
    test("<Navbar/> matches snapshot", () => {
        const component = render(<Navbar />);
        expect (component.container).toMatchSnapshot();
    });

});