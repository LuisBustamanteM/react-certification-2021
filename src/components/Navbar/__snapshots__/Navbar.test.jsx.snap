// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Navbar/> <Navbar/> matches snapshot 1`] = `
<div>
  <div
    class="sc-dQoVA ckayCX"
  >
    <div
      class="sc-eCstlR jhFGSs"
    >
      <div
        class="sc-bdfBQB hchZau"
      >
        <div
          class="sc-gsTEea gextng"
        >
          <div
            class="sc-dlfnuX bxyajH"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
          </div>
          <input
            class="sc-hKgJUU gfleLt"
            placeholder="Search..."
            type="text"
            value="Wizeline"
          />
        </div>
      </div>
    </div>
    <div
      class="sc-crrszt hatZDm"
    >
      <div
        class="sc-jSgvzq ctEpNB"
      >
        <div
          class="sc-fubCzh ehNjeG"
        >
          <input
            checked=""
            type="checkbox"
          />
          <div
            class="sc-pGacB clwlSv slider"
            role="slider"
          />
        </div>
        <label
          class="sc-gKseQn hzDtly"
          for="toggle-mode"
        >
          Dark
           Mode
        </label>
      </div>
      <div
        class="sc-jrAFXE clIODa"
      >
        <label
          class="sc-kEjbQP rNvkg"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
            />
          </svg>
        </label>
      </div>
    </div>
  </div>
</div>
`;
